src/Misc/WaveShapeSmps.cpp

```diff
@@ -16,20 +16,64 @@
 
 namespace zyn {
 
+float polyblampres(float smp, float ws, float dMax)
+{
+    // Formula from: Esqueda, Välimäki, Bilbao (2015): ALIASING REDUCTION IN SOFT-CLIPPING ALGORITHMS
+    // http://dafx16.vutbr.cz/dafxpapers/18-DAFx-16_paper_33-PN.pdf pg 123, table 1 
+    // Four-point polyBLAMP residual:
+    // [−2T, T] d^5/120
+    // [−T, 0] −d^5/40 + d^4/24 + d^3/12 + d^2/12 + d/24 + 1/120
+    // [0, T] d^5/40 − d^4/12 + d^2/3 − d/2 + 7/30
+    // [T, 2T] −d^5/120 + d^4/24 − d^3/12 + d^2/12 − d/24 + 1/120
+
+    float dist = fabs(smp) - ws;
+    float res, d;
+    if (fabs(dist) < dMax) {
+        if (dist < -dMax/2.0f) {
+            d = (dist + dMax)/dMax*2;   // [-dMax ... -dMax/2] -> [0 ... 1]
+            res = powf(d, 5.0f) / 120.0f;
+        }
+        else if ( dist < 0.0) {
+            d = (dist + dMax/2)/dMax*2; // [-dMax/2 ... 0] -> [0 ... 1]
+            res = -powf(d,5.0f)/40.0f + powf(d,4.0f)/24.0f + powf(d,3.0f)/12.0f + powf(d,2.0f)/12.0f + d/24.0f + 1.0f/120.0f;
+        }
+        else if ( dist < dMax/2.0) {
+            d = (dist)/dMax*2;          //[0 ... dMax/2] -> [0 ... 1]
+            res = (powf(d,5.0f)/40.0f) - (powf(d,4.0)/12.0f) + (powf(d,2.0f)/3.0f) - (d/2.0f) + (7.0f/30.0f);
+        }
+        else {
+            d = (dist - dMax/2.0)/dMax*2; //[dMax/2 ... dMax] -> [0 ... 1]
+            res = -powf(d,5.0f)/120.0f + powf(d,4.0f)/24.0f - powf(d,3.0f)/12.0f + powf(d,2.0f)/12.0f - d/24.0f + 1.0f/120.0f;
```

Comment:
Instead of calling `powf` four times, will it be more efficient to iteratively compute power by power?

Reply:
goot idea